### ISSUE TITLE
Expose *STM functions

### ISSUE THREAD

**Issue by charles-cooper** [hverr/haskell-cache#5]
As an API consumer building on top of this library it would be useful to have access to the underlying STM functions. For instance, an implementation for inserting a key if there is 'enough space' might look like follows:
```
insertIfSpace :: Int -> k -> v -> Cache k v -> Maybe TimeSpec -> STM ()
insertIfSpace maxSize k v cache mexpiry = do
  t <- now
  atomically $ do
    -- purge expired keys to free up space
    ks <- keysSTM cache
    forM_ ks $ \k -> lookupItemT True k cache t

    -- insert if doing so would not make the cache too big
    sz <- sizeSTM cache
    when (sz < maxSize) $ do
      insertT k v cache mexpiry
```
Unfortunately only having access to the IO functions means that this can't be implemented in a transactionally safe manner.

**Comment by hverr** [comment]
Sure, no problem, will do when I have time (i.e. Saturday or Sunday). Feel free to submit a PR!

**Comment by hverr** [comment]
@charles-cooper Could you review PR #6?

Note that in your code example, after merging PR #6, you can write

```
    purgeExpiredSTM cache now
```

instead of 

```
    -- purge expired keys to free up space
    ks <- keysSTM cache
    forM_ ks $ \k -> lookupItemT True k cache now
```

**Comment by hverr** [comment]
Solved in [cache-0.1.1.0](https://hackage.haskell.org/package/cache-0.1.1.0)
